tailscale/rootfs/etc/s6-overlay/s6-rc.d/proxy/run

```diff
@@ -31,6 +31,13 @@ if bashio::var.true "$(bashio::core.ssl)"; then
   bashio::exit.nok
 fi
 
+# Test Home Assistant's HTTP reverse proxy configuration
+if (( 200 != $(curl -s -o /dev/null -w "%{http_code}" "http://127.0.0.1:$(bashio::core.port)" -H "X-Forwarded-For: 127.0.0.1") )); then
```

Comment:
Home Assistant doesn't have to be running?

Reply:
It tests that the `X-Forwarded-For: 127.0.0.1` is added to the HA configuration. People usually forget it and then see the "400: Bad Request" error and complain. so I've made a test and a log entry.

**It has a "dangerous" side effect:** if by default the proxy is enabled, but the user has never tried it and never noticed the misconfiguration, the add-on started previously, but with this change, it won't start, because the default enabled proxy is not working, so it will brake some systems, this happened once with one of "my" forked repo's users.

And it also means, the after a fresh add-on install, the add-on won't start, users have to reconfigure HA or disable the TS proxy. It's a bit strange, but follows your concept, to fail if there is something misconfigured.

Reply:
I get what it does and why you did it, but there is no guarantee Home Assistant is even running when it reaches this point.

Reply:
Oh, I see. `startup: services` will start before Home Assistant, though nobody run into this problem yet, the add-on starts up much slower then HA core.

Maybe emit the error, or only a warning, but do not `exit.nok`?

Reply:
A little bit of a d-tour, maybe, but I can expose the core state so we can check if Core is running or not. This might help with the decision to run this check.

I can probably make this happen friday/the weekend.

Reply:
That sounds great! If HA runs, current behavior (error), if doesn't, only a warning about it.

Reply:
Updated the base image, which contains a new method `bashio::info.state`, if the output of that equals to `running` you'd know Core is up!

Reply:
Isn't that the Supervisor's state? It is constant "running", API docs says: "The core state of the Supervisor."

That field returns `self.sys_core.state`

We would need sg. like `self.sys_homeassistant.core.is_running`, but I can't find it on the API.

Reply:
> Isn't that the Supervisor's state?

You are so right :(

"core" in the Supervisor API threw me off. Will check with the team.

Reply:
Made a workaround: first curl HA **without** `-H "X-Forwarded-For: ...` to test whether it is running (accessible) or not, if running, test with `-H "X-Forwarded-For: ...`

Tested, all 3 cases working:
- core is not runnng
- core is running without proper config
- core is running with proper config

Rebased, ready to review.